File: banchus-packet/src/main/java/pe/nanamochi/banchus/components/Mode.java
package pe.nanamochi.banchus.components;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum Mode {
  OSU(0),
  TAIKO(1),
  CATCH(2),
  MANIA(3);

  private final int value;

  public static Mode fromValue(int value) {
    for (Mode mode : values()) {
      if (mode.value == value) {
        return mode;
      }
    }
    return OSU;
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/components/Mode.kt
package pe.nanamochi.banchus.components

enum class Mode(val value: UByte) {
    OSU(0u),
    TAIKO(1u),
    CATCH(2u),
    MANIA(3u);

    companion object {
        fun fromValue(value: UByte): Mode = entries.find { it.value == value } ?: OSU
    }
}
